onnx/onnx-ml.proto

```diff
@@ -384,10 +384,39 @@ message TensorShapeProto {
       int64 dim_value = 1;
       string dim_param = 2;   // namespace Shape
     };
+    // Standard denotation can optionally be used to denote tensor
+    // dimensions with standard semantic descriptions to ensure
+    // that operations are applied to the correct axis of a tensor.
+    optional string standard_denotation = 3;
```

Comment:
what do you think of calling it "semantic_denotation"?

potentially it might be useful also with non-standard names

Reply:
OK, I see this might warrant the ad-hoc situation you mentioned above. Sounds reasonable.